src/pudl/transform/eia861.py

```diff
@@ -194,79 +615,708 @@ def service_territory(raw_dfs, tfr_dfs):
+
+    """
+    return tfr_dfs
+
+
+##############################################################################
+# Coordinating Transform Function
+##############################################################################
+
+def transform(raw_dfs, eia861_tables=pc.pudl_tables["eia861"]):
+    """
+    Transform EIA 861 DataFrames.
 
     Args:
         raw_dfs (dict): a dictionary of tab names (keys) and DataFrames
             (values). This can be generated by pudl.
-        tables (tuple): A tuple containing the names of the EIA 861
+        eia861_tables (tuple): A tuple containing the names of the EIA 861
             tables that can be pulled into PUDL
 
     Returns:
-        dict: A dictionary of DataFrame objects in which pages from EIA861 form
+        dict: A dictionary of DataFrame objects in which pages from EIA 861 form
         (keys) corresponds to a normalized DataFrame of values from that page
         (values)
 
     """
     # these are the tables that we have transform functions for...
-    eia861_transform_functions = {
-        'service_territory_eia861': service_territory,
+    tfr_funcs = {
+        "balancing_authority_eia861": balancing_authority,
+        "service_territory_eia861": service_territory,
+        "sales_eia861": sales,
+        "advanced_metering_infrastructure_eia861": advanced_metering_infrastructure,
+        "demand_response_eia861": demand_response,
+        # "demand_side_management_eia861": demand_side_management,
+        # "distributed_generation_eia861": distributed_generation,
+        "distribution_systems_eia861": distribution_systems,
+        "dynamic_pricing_eia861": dynamic_pricing,
+        "green_pricing_eia861": green_pricing,
+        # "mergers_eia861": mergers,
+        # "net_metering_eia861": net_metering,
+        # "non_net_metering_eia861": non_net_metering,
+        # "operational_data_eia861": operational_data,
+        # "reliability_eia861": reliability,
+        # "utility_data_eia861": utility_data,
     }
     tfr_dfs = {}
 
     if not raw_dfs:
-        logger.info("No raw EIA 861 dataframes found. "
-                    "Not transforming EIA 861.")
+        logger.info(
+            "No raw EIA 861 dataframes found. Not transforming EIA 861.")
         return tfr_dfs
     # for each of the tables, run the respective transform funtction
-    for table in eia861_transform_functions:
-        if table in tables:
-            logger.info(f"Transforming raw EIA 861 DataFrames for {table} "
-                        f"concatenated across all years.")
-            eia861_transform_functions[table](raw_dfs, tfr_dfs)
+    for table in eia861_tables:
+        logger.info(f"Transforming raw EIA 861 DataFrames for {table} "
+                    f"concatenated across all years.")
+        # ADD some sort of customer message here / check across multiple sources
+        assert table in tfr_funcs.keys()
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/zaneselvans/pudl/pullRequest?prid=5759597)